src/algorithms/calorimetry/ImagingTopoCluster.cc

```diff
@@ -50,48 +51,103 @@ void ImagingTopoCluster::init() {
+      throw std::runtime_error(msg);
+    }
+    info("Same-layer clustering (same sector and same layer): "
+         "Global [eta, phi] distance between hits <= [{:.4f}, {:.4f} rad].",
+         sameLayerDistEtaPhi[0], sameLayerDistEtaPhi[1]);
+    break;
+  case ImagingTopoClusterConfig::ELayerMode::tz:
+    if (m_cfg.sameLayerDistTZ.size() != 2) {
+      const std::string msg = "Expected 2 values (phi_dist, z_dist) for sameLayerDistTZ";
+      error(msg);
+      throw std::runtime_error(msg);
+    }
+    info("Same-layer clustering (same sector and same layer): "
+         "Global [phi, z] distance between hits <= [{:.4f} rad, {:.4f} mm].",
+         sameLayerDistTZ[0], sameLayerDistTZ[1]);
+    break;
+  default:
+    throw std::runtime_error("Unknown same-layer mode.");
+  }
+
+  // different layer clustering parameters
+  switch (m_cfg.diffLayerMode) {
   case ImagingTopoClusterConfig::ELayerMode::etaphi:
-    info("Neighbour layers clustering (same sector and layer id within +- {:d}: "
+    if (m_cfg.diffLayerDistEtaPhi.size() != 2) {
+      const std::string msg = "Expected 2 values (eta_dist, phi_dist) for diffLayerDistEtaPhi";
+      error(msg);
+      throw std::runtime_error(msg);
+    }
+    info("Neighbour layers clustering (same sector and layer id within +- {:d}): "
          "Global [eta, phi] distance between hits <= [{:.4f}, {:.4f} rad].",
-         m_cfg.neighbourLayersRange, layerDistEtaPhi[0], layerDistEtaPhi[1]);
+         m_cfg.neighbourLayersRange, diffLayerDistEtaPhi[0], diffLayerDistEtaPhi[1]);
     break;
   case ImagingTopoClusterConfig::ELayerMode::xy:
-    info("Neighbour layers clustering (same sector and layer id within +- {:d}: "
-         "Local [x, y] distance between hits <= [{:.4f} mm, {:.4f} mm].",
-         m_cfg.neighbourLayersRange, layerDistXY[0], layerDistXY[1]);
+    if (m_cfg.diffLayerDistXY.size() != 2) {
+      const std::string msg = "Expected 2 values (x_dist, y_dist) for diffLayerDistXY";
+      error(msg);
+      throw std::runtime_error(msg);
+    }
+    info("Neighbour layers clustering (same sector and layer id within +- {:d}): "
+         "Global [x, y] distance between hits <= [{:.4f} mm, {:.4f} mm].",
+         m_cfg.neighbourLayersRange, diffLayerDistXY[0], diffLayerDistXY[1]);
+    break;
+  case ImagingTopoClusterConfig::ELayerMode::tz:
+    if (m_cfg.diffLayerDistTZ.size() != 2) {
+      const std::string msg = "Expected 2 values (phi_dist, z_dist) for diffLayerDistTZ";
+      error(msg);
+      throw std::runtime_error(msg);
+    }
+    info(
+        "Neighbour layers clustering (same sector and layer id within +- {:d}): "
+        "Global [phi, z] distance between hits <= [{:.4f} mm, {:.4f} mm].", // The coordinate Phi is the projected Phi and thus is a distance, not an angle.
+        m_cfg.neighbourLayersRange, diffLayerDistTZ[0], diffLayerDistTZ[1]);
     break;
   default:
-    error("Unknown layer mode.");
+    throw std::runtime_error("Unknown different-layer mode.");
```

Comment:
Keep the explicit error logging, since not all runtime error exceptions lead to a printout.
```suggestion
    error("Unknown different-layer mode.");
    throw std::runtime_error("Unknown different-layer mode.");
```